internal/provider/resource_group.go

```diff
@@ -118,19 +123,17 @@ func resourceGroupCreate(ctx context.Context, d *schema.ResourceData, meta inter
 		}
 		gcsmr, err := grps.SetMembers(ctx, gcr.Item.Id, gcr.Item.Version, memberIds)
 		if err != nil {
-			return diag.Errorf("error setting principals on role: %v", err)
+			errs = append(errs, diag.Errorf("error setting members on group: %v", err)...)
+			return errs
```

Comment:
This shouldn't need to change, just returning will do the assignment, and the variable will be mutable in the deferred function
```suggestion
			return diag.Errorf("error setting members on group: %v", err)
```